### PR TITLE
feat: Requests and Events

### PR DESCRIPTION
**Pull request by ChristopheCVB**
Upgraded to Gradle 6.8.3
Upgraded Gson dependency to latest version
A bit of Cleanup
Added some Requests and Events
Refactor Events

### CODE REVIEW

**Comment by TinaTiel** [comment]
Thanks! Taking a look...

File: src/main/java/net/twasi/obsremotejava/objects/Scene.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/objects/Scene.java:18]: Could you add javadocs here explaining the difference between these two methods? 
Or rename it to something like "getSourcesIncludingGroupChildren"?

File: src/main/java/net/twasi/obsremotejava/OBSCommunicator.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/OBSCommunicator.java:129]: Awesome! Closes #27

**Review by TinaTiel — COMMENTED** [review]
I've reviewed the files and will try testing locally next...Only request so far was some more javadocs on one place, otherwise looks fantastic so far!

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:246]: Could you add this to ObsRemoteControllerUnsecuredIT#test?

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:397]: Getting an "invalid request type" message when using this, can you repro?

onMessage: {
    "error": "invalid request type",
    "message-id": "4",
    "status": "error"
}

**Review by TinaTiel — CHANGES_REQUESTED** [review]
A bit more review, requesting some changes on one thing so far. 
Otherwise testing was good, and I'll post a message to the original maintainers asking for their final feedback / if we can merge after the requested changes.

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:407]: should be false

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (TinaTiel) [inline src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:411]: should be true

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (ChristopheCVB) [reply src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:407]: Nice catch, sorry ;)

File: src/main/java/net/twasi/obsremotejava/OBSRemoteController.java
Comment (TinaTiel) [reply src/main/java/net/twasi/obsremotejava/OBSRemoteController.java:251]: Tested and confirmed!

**Review by TinaTiel — APPROVED** [review]
Looks great! I approve!
Still pending response from owners, I'll ping them and if there's no response in a few days then merge if that's fine with you.

**Review by LarsBaertschi — APPROVED** [review]
Thanks for this awesome extension! Looks good to me, we'll make sure to get this released once it's merged as new minor version!

**Comment by ChristopheCVB** [comment]
> Thanks for this awesome extension! Looks good to me, we'll make sure to get this released once it's merged as new minor version!

Thanks for the kind words.
Fine by me ;)

**Comment by ChristopheCVB** [comment]
I'm sorry guys, there were things that bothered me. I went ahead and refactored a bit.
No change to the public API as long as developers use lambdas.

File: build.gradle
Comment (ChristopheCVB) [inline build.gradle:24]: jcenter is deprecated

**Comment by TinaTiel** [comment]
Ran the tests again, all seems well.

**Comment by ChristopheCVB** [comment]
> Ran the tests again, all seems well.

I added some too and ran them locally before pushing each commit, including the OBSRemoteControllerUnsecuredIT

**Pull request merged**
